### ISSUE TITLE
Memory leak in localhost mode

### ISSUE THREAD

**Issue by raoz** [splitio/javascript-commons#181]
When running in localhost mode, the impressions cache seems to grow unboundedly, as the following condition never triggers: https://github.com/splitio/javascript-commons/blob/development/src/storages/inMemory/ImpressionsCacheInMemory.ts#L31 because `onFullQueue` is never set.

I was pointed there by heap profiler, as I was investigating an out of memory error in a development pipeline with a long-running analysis task (which uses split in localhost mode).

**Comment by hbqdev** [comment]
HI @raoz 
We are reviewing this issue with our engineering team.

Thank you and Regards

**Comment by EmilianoSanchez** [comment]
Hi @raoz ,

As you mentioned, `onFullQueue` callback is never set in localhost mode, leading to memory usage growing unboundedly.

We have released a new version of [@splitsoftware/splitio-commons](https://www.npmjs.com/package/@splitsoftware/splitio-commons) with a fix, and upgraded it in our [Javascript SDK v10.22.3](https://www.npmjs.com/package/@splitsoftware/splitio/v/10.22.3). The approach we followed is to not keep in memory the events and impressions generated by calling the `track` and `getTreatment*` methods, if the SDK factory is in localhost mode.

 Thank you
